src/tree/tree.rs

```diff
@@ -0,0 +1,223 @@
+use super::{
+    PoseidonBranch, PoseidonLeaf, PoseidonTreeAnnotation,
+    PoseidonWalkableAnnotation,
+};
+use canonical::{Canon, Store};
+use canonical_derive::Canon;
+use dusk_bls12_381::BlsScalar;
+use microkelvin::{Branch, Cardinality, Nth};
+use nstack::NStack;
+
+/// Represents a Merkle Tree with a given depth that will be calculated using poseidon hash
+///
+/// The `BlsScalar` borrow of the annotation must represent the root poseidon merkle opening
+/// for the annotated subtree
+#[derive(Debug, Clone, Canon)]
+pub struct PoseidonTree<L, A, S, const DEPTH: usize>
```

Comment:
I still think this was OK inside of `mod.rs` although I understand that moving it to a sepparate module makes dealing with features easier.

Fine for me, but it might end up in an API change. Not sure. any thoughts @vlopes11 ?